hack/jenkins/upload-to-gcs.sh

```diff
@@ -40,19 +54,78 @@ readonly artifacts_path="${WORKSPACE}/_artifacts"
 readonly gcs_job_path="${JENKINS_GCS_LOGS_PATH}/${JOB_NAME}"
 readonly gcs_build_path="${gcs_job_path}/${BUILD_NUMBER}"
 readonly gcs_acl="public-read"
+readonly results_url=${gcs_build_path//"gs:/"/"https://storage.cloud.google.com"}
+readonly timestamp=$(date +%s)
+
+function upload_version() {
+  echo -n 'Run starting at '; date -d "@${timestamp}"
 
-for upload_attempt in $(seq 3); do
-  echo "Uploading to ${gcs_build_path} (attempt ${upload_attempt})"
-  if [[ -d "${artifacts_path}" && -n $(ls -A "${artifacts_path}") ]]; then
-    gsutil -m -q -o "GSUtil:use_magicfile=True" cp -a "${gcs_acl}" -r -c \
-      -z log,txt,xml "${artifacts_path}" "${gcs_build_path}/artifacts" || continue
+  # Try to discover the kubernetes version.
+  if [[ -e "version" ]]; then
+    version=$(cat "version")
+  elif [[ -e "hack/lib/version.sh" ]]; then
+    version=$(
+      export KUBE_ROOT="."
+      source "hack/lib/version.sh"
+      kube::version::get_version_vars
+      echo "${KUBE_GIT_VERSION-}"
+    )
   fi
-  # Mark this build as the latest completed.
-  echo "${BUILD_NUMBER}" | \
-    gsutil -q -h "Content-Type:text/plain" -h "Cache-Control:private, max-age=0, no-transform" \
-      cp -a "${gcs_acl}" - "${gcs_job_path}/latest-build.txt" || continue
-  break  # all uploads succeeded if we hit this point
-done
 
-readonly results_url=${gcs_build_path//"gs:/"/"https://storage.cloud.google.com"}
-echo -e "\n\n\n*** View logs and artifacts at ${results_url} ***\n\n"
+  if [[ -n "${version}" ]]; then
```

Comment:
I think this will break if `version` is unset. Previously I was setting it to `""` before trying to discover the version.